dubbo-config/dubbo-config-spring/src/main/java/com/alibaba/dubbo/config/spring/AnnotationBean.java

```diff
@@ -96,6 +96,14 @@ public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
                 // spring 2.0
             }
         }
+
+
+        //
+        if(!beanFactory.containsBean(DynamicReferenceService.class.getName())) {
+            //注册服务bean
+            beanFactory.registerSingleton(DynamicReferenceService.class.getName(), new DynamicReferenceService(this));

```

Comment:
Why do you need to hard code the injected object ？ Do not allow multiple DynamicReferenceServices and different ids ? Or my understanding is wrong?

Reply:
DynamicReferenceService is like a util class. It is the entrance to acquire a Reference service dynamic . like blow: 
    

SomeDubboService service = dynamicReferenceService.getDubboService(SomeDubboService.class,rc);